DBB_DOCS.md

```diff
@@ -0,0 +1,180 @@
+   ```
+
+#. (Optional) Create the DynamoDB table yourself.
+
+A single DynamoDB table will maintain commit metadata for multiple Delta tables, and it is important that your DynamoDB table is configured with sufficient [Read and Write Capacity Units](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadWriteCapacityMode.html#HowItWorks.ProvisionedThroughput.Manual).
+
+If the default DynamoDB table does not already exist then it will be created for you automatically. This default table supports 5 strongly consistent reads and 5 writes per second. You may change these default values using the configurations below:
+
+   ```
+   - spark.delta.DynamoDBLogStore.provisionedThroughput.rcu - Read Capacity Units (defaults to 5)
+   - spark.delta.DynamoDBLogStore.provisionedThroughput.wcu - Write Capacity Units (defaults to 5)
+   ```
+
+Alternatively, you can create the DynamoDB table yourself. An example is provided below using the AWS CLI. Learn more [here](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/getting-started-step-1.html).
+
+   ```bash
+   aws \
+     --region us-west-2 dynamodb create-table \
+     --table-name delta_log \
+     --attribute-definitions AttributeName=tablePath,AttributeType=S \
+                             AttributeName=fileName,AttributeType=S \
+     --key-schema AttributeName=tablePath,KeyType=HASH \
+                   AttributeName=fileName,KeyType=RANGE \
+     --provisioned-throughput ReadCapacityUnits=5,WriteCapacityUnits=5
+   ```
+
+#. Cleanup old DynamoDB entries using Time to Live (TTL).
+
+Once a DynamoDB metadata entry is marked as complete, and after sufficient time such that we can now rely on S3 alone to prevent accidental overwrites on the same Delta file, it is safe to delete that entry from DynamoDB. The cheapest way to do this is using [DynamoDB's TTL](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/TTL.html) feature which is a free, automated means to delete items from your table.
+
+Run the following command on your given DynamoDB table to enable TTL:
+
+   ```bash
+   aws dynamodb \
+     --region us-west-2 update-time-to-live \
+     --table-name delta_log \
+     --time-to-live-specification "Enabled=true, AttributeName=commitTime"
+   ```
+
+#. Cleanup old AWS S3 temp files using S3 Lifecycle Expiration.
+
+In this `LogStore` implementation, a temp file is created containing a copy of the metadata to be committed into the Delta log. Once that commit to the Delta log is complete, and after the corresponding DynamoDB entry has been removed, it is safe to delete this temp file.
+
+We suggest deleting temp files using an [S3 Lifecycle Expiration rule](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html), with filter prefix pointing to the `/_delta_log/.tmp` folder located in your table path, and an expiration value of 3 days.
```

Comment:
I think we need to be very careful here - for some particular data access patterns we may remove temporary file for incomplete dynamodb entry (and we are suggesting quite short expiration period here).

Reply:
We only ever need the latest temp file in the `.tmp` folder. Why do you think that 3 days is short?

The only way this can fail is: last commit failed, and we do not read or write for 3 days.

What duration do you suggest?

Reply:
I can imagine failure of job writing data (so writing is stopped), job is fixed after few days but in the meantime temporary file for last incomplete entry expires. 
I cannot suggest any duration because I don't like the whole idea - increasing duration will make data loss less probable, but still possible.

Maybe instead we should design some AWS lambda function removing these temporary files smart way (only if corresponding dynamodb entries do not exist / are complete)? Or maybe clean these files during vacuum some way?

Reply:
cc @tdas 

We have a few options
- suggest expiration value of, say, 30 days, and inform the user of the risks
- suggest a simple script to remove all `.tmp` files except the latest, e.g. something like this:
```
aws s3 ls s3://somebucket/ --recursive | sort | head -n -1 | while read -r line ; do
    echo "Removing ${line}"
    aws s3 rm s3://somebucket/${line}
done

Note: i have not reviewed the cost impacts of this ...
```
- [Future Work] proper delete implementation in code
- [Future Work] AWS Lambda function?

Reply:
why not both? no need to say future work. just create an issue.